### PR TITLE
Fix saving translations when only translations are modified.

### PR DESCRIPTION
**Pull request by markstory**
When we have bundled translations but no changed fields, we need to mark the original fields dirty so we can save the root entity and the attached translations.

Refs #9610

### CODE REVIEW

File: src/ORM/Behavior/TranslateBehavior.php [hunk 1]
```diff
@@ -271,22 +271,35 @@ public function beforeSave(Event $event, EntityInterface $entity, ArrayObject $o
...
-        $primaryKey = (array)$this->_table->primaryKey();
-        $key = $entity->get(current($primaryKey));
-        $model = $this->_config['referenceName'];
+        // If there no fields, and bundled translations,
+        // we need to mark fields as dirty so the entity persists.
+        if ($noFields && $bundled) {
+            foreach ($this->_config['fields'] as $field) {
+                $entity->dirty($field, true);
+            }
+            return;
```
Comment (thinkingmedia) [inline src/ORM/Behavior/TranslateBehavior.php]: 🙄

File: src/ORM/Behavior/TranslateBehavior.php [hunk 2]
```diff
@@ -271,22 +271,36 @@ public function beforeSave(Event $event, EntityInterface $entity, ArrayObject $o
...
+        // as the entity is in the default locale.
+        if ($noBundled && $locale === $this->config('defaultLocale')) {
             return;
         }
 
         $values = $entity->extract($this->_config['fields'], true);
         $fields = array_keys($values);
+        $noFields = empty($fields);
+
+        $primaryKey = (array)$this->_table->primaryKey();
```
Comment (lorenzo) [inline src/ORM/Behavior/TranslateBehavior.php]: Why move these lines up here?

Comment (markstory) [inline src/ORM/Behavior/TranslateBehavior.php]: I can move them back down.

**Comment by markstory** [comment]
I'll look at the postgres builds tonight.

**Comment by lorenzo** [comment]
Great job!

**Pull request merged**